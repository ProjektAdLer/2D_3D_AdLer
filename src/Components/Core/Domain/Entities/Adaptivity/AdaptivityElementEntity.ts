import { AdaptivityElementQuestionDifficultyTypes } from "../../Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import LearningElementEntity from "../LearningElementEntity";

export default class AdaptivityElementEntity {
  element: LearningElementEntity;
  introText: string;
  shuffleTask: boolean;
  tasks: AdaptivityTaskEntity[];
}

export class AdaptivityTaskEntity {
  taskId: number;
  taskTitle: string;
  taskOptional: boolean;
  requieredDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questions: AdaptivityQuestionEntity[];
}

export class AdaptivityQuestionEntity {
  questionType: string;
  questionId: number;
  questionDifficulty: AdaptivityElementQuestionDifficultyTypes;
  questionText: string;
  questionAnswers: AdaptivityAnswerEntity[];
  trigger: AdaptivityTriggerEntity[];
}

export class AdaptivityAnswerEntity {
  answerId: number;
  answerText?: string;
  answerImage?: string;
}

export class AdaptivityTriggerEntity {
  triggerId: number;
  triggerType: string;
  triggerCondition: string;
  triggerAction: AdaptivityActionEntity;
}

export class AdaptivityActionEntity {
  actionType: string;
  actionData: string | number;
}
