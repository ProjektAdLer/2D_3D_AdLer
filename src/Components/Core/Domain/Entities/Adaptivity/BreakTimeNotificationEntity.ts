export enum BreakTimeNotificationType {
  None = 0,
  Short = 5,
  Medium = 15,
  Long = 30,
}

export default class BreakTimeNotificationEntity {
  breakTimeIntervalCounter: number;
}
