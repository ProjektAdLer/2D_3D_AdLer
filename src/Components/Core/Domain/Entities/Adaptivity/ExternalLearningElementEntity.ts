import { ComponentID } from "../../Types/EntityTypes";
import { LearningElementTypeStrings } from "../../Types/LearningElementTypes";

export default class ExternalLearningElementEntity {
  id: ComponentID;
  worldID: ComponentID;
  name: string;
  type: LearningElementTypeStrings;
}
