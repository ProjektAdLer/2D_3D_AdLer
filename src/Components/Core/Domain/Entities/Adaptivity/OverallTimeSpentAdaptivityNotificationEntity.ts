export enum OverallTimeSpentAdaptivityNotificationBreakType {
  None = 0,
  Short = 30,
  Medium = 60,
  Long = 90,
}

export enum OverallTimeSpentAdaptivityNotificationWaitingType {
  Short = 5,
  Medium = 15,
  Long = 30,
}

export default class OverallTimeSpentAdaptivityNotificationEntity {
  notificationType: OverallTimeSpentAdaptivityNotificationBreakType;
}
