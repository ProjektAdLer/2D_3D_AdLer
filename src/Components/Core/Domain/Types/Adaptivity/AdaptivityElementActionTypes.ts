// TODO: add string representations
export enum AdaptivityElementActionTypes {
  CommentAction = "CommentAction",
  ReferenceAction = "AdaptivityReferenceAction",
  ContentAction = "AdaptivityContentReferenceAction",
}
