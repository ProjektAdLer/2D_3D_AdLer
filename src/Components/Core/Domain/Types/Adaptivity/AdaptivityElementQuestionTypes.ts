export enum AdaptivityElementQuestionTypes {
  singleChoice = "SingleChoice",
  multipleChoice = "MultipleChoice",
}
