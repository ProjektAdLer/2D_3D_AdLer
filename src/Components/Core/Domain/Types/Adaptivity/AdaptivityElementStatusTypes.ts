export enum AdaptivityElementStatusTypes {
  correct = "Correct",
  incorrect = "Incorrect",
}
