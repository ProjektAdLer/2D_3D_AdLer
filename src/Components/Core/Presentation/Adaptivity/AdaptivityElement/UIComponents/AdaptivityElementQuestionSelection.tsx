import { useEffect } from "react";
import {
  AdaptivityHint,
  AdaptivityQuestion,
  AdaptivityTask,
} from "../AdaptivityElementViewModel";
import StyledButton from "~ReactComponents/ReactRelated/ReactBaseComponents/StyledButton";
import { AdaptivityElementQuestionDifficultyTypes } from "src/Components/Core/Domain/Types/Adaptivity/AdaptivityElementQuestionDifficultyTypes";
import requiredTaskIcon from "../../../../../../Assets/icons/41-required-adaptivity/required-adaptivity.svg";
import AdaptivityElementDifficultyStars, {
  AdaptivityElementDifficultyStarState,
} from "./AdaptivityElementDifficultyStars";
import { useTranslation } from "react-i18next";

export default function AdaptivityElementQuestionSelection({
  selectedTask,
  setHeaderText,
  onSelectQuestion,
  onSelectHint,
}: {
  selectedTask: AdaptivityTask;
  setHeaderText: (headerText: string) => void;
  onSelectQuestion: (selectedQuestion: AdaptivityQuestion) => void;
  onSelectHint: (
    selectedHint: AdaptivityHint,
    associatedQuestion: AdaptivityQuestion
  ) => void;
}) {
  const { t: translate } = useTranslation("learningElement");

  useEffect(() => {
    setHeaderText(
      translate("taskSelectionTitle", {
        selectedTaskTitle: selectedTask.taskTitle,
      })
    );
  }, [setHeaderText, selectedTask, translate]);

  return (
    <div className="grid w-full gap-4 px-2 py-2">
      {selectedTask.questions.map((question) => {
        let starState = AdaptivityElementDifficultyStarState.RequiredUnsolved;
        if (question.isCompleted === true) {
          starState = AdaptivityElementDifficultyStarState.RequiredSolved;
        } else if (question.isCompleted === false) {
          starState = AdaptivityElementDifficultyStarState.RequiredTried;
        }

        return (
          <div
            key={question.questionID}
            className="grid items-center justify-center w-full grid-cols-3 gap-4 md:grid-cols-2"
          >
            <StyledButton
              shape="freefloatcenter"
              className="w-full col-span-2 md:col-span-1"
              onClick={() => onSelectQuestion(question)}
            >
              <div className="grid items-center w-full h-full grid-cols-5 ">
                {question.difficulty ===
                  AdaptivityElementQuestionDifficultyTypes.easy && (
                  <>
                    <AdaptivityElementDifficultyStars
                      easyState={starState}
                      starClassName="w-4 h-4 md:w-6 md:h-6"
                    />
                    <p className="flex justify-center col-span-3 px-4 text-sm text-start">
                      {translate("easy")}{" "}
                    </p>
                  </>
                )}
                {question.difficulty ===
                  AdaptivityElementQuestionDifficultyTypes.medium && (
                  <>
                    <AdaptivityElementDifficultyStars
                      mediumState={starState}
                      starClassName="w-4 h-4 md:w-6 md:h-6"
                    />
                    <p className="flex justify-center col-span-3 px-4 text-sm text-start">
                      {translate("normal")}{" "}
                    </p>
                  </>
                )}
                {question.difficulty ===
                  AdaptivityElementQuestionDifficultyTypes.hard && (
                  <>
                    <AdaptivityElementDifficultyStars
                      hardState={starState}
                      starClassName="w-4 h-4 md:w-6 md:h-6"
                    />
                    <p className="flex justify-center col-span-3 px-4 text-sm text-start">
                      {translate("hard")}{" "}
                    </p>
                  </>
                )}
                <div className="flex items-center justify-end pr-2">
                  {question.isRequired && (
                    <img
                      alt=""
                      className={"h-5 md:h-7"}
                      src={requiredTaskIcon}
                    />
                  )}

                  {!question.isRequired && <div className="w-16"></div>}
                </div>
              </div>
            </StyledButton>
            <div className="flex flex-row h-full md:gap-4 ">
              {question.hints.length > 0 &&
                question.hints.map((hint) => {
                  if (
                    (hint.showOnIsWrong && question.isCompleted === false) ||
                    (!hint.showOnIsWrong && question.isCompleted === true)
                  ) {
                    return (
                      <StyledButton
                        shape="freefloatcenter"
                        className="w-full h-full "
                        key={"hint-" + hint.hintID}
                        onClick={() => onSelectHint(hint, question)}
                      >
                        <p className="text-sm">Hinweis</p>
                      </StyledButton>
                    );
                  } else return null;
                })}
            </div>
          </div>
        );
      })}
    </div>
  );
}
