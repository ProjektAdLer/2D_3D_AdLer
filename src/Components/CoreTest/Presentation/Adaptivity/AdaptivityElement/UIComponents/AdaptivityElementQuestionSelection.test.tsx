import { render } from "@testing-library/react";
import AdaptivityElementQuestionSelection from "../../../../../Core/Presentation/Adaptivity/AdaptivityElement/UIComponents/AdaptivityElementQuestionSelection";
import React from "react";
import { AdaptivityElementActionTypes } from "../../../../../Core/Domain/Types/Adaptivity/AdaptivityElementActionTypes";

describe("AdaptivityElementQuestionSelection", () => {
  // ANF-ID: [EWE0012]
  test("should render (with all difficulties)", () => {
    const { container } = render(
      <AdaptivityElementQuestionSelection
        selectedTask={{
          taskID: 0,
          taskTitle: "testTaskTitle",
          isRequired: false,
          isCompleted: false,
          requiredDifficulty: 0,
          questions: [
            {
              questionID: 0,
              questionText: "testQuestionText",
              isMultipleChoice: false,
              difficulty: 0,
              isCompleted: null,
              isRequired: false,
              questionAnswers: [
                {
                  answerIndex: 0,
                  answerText: "testAnswerText",
                  isSelected: false,
                },
              ],
              hints: [
                {
                  hintID: 0,
                  showOnIsWrong: true,
                  hintAction: {
                    hintActionData: "testHintActionData",
                    hintActionType: AdaptivityElementActionTypes.CommentAction,
                  },
                },
              ],
            },
            {
              questionID: 0,
              questionText: "testQuestionText",
              isMultipleChoice: false,
              difficulty: 100,
              isCompleted: true,
              isRequired: false,
              questionAnswers: [
                {
                  answerIndex: 0,
                  answerText: "testAnswerText",
                  isSelected: false,
                },
              ],
              hints: [
                {
                  hintID: 0,
                  showOnIsWrong: false,
                  hintAction: {
                    hintActionData: "testHintActionData",
                    hintActionType: AdaptivityElementActionTypes.CommentAction,
                  },
                },
              ],
            },
            {
              questionID: 0,
              questionText: "testQuestionText",
              isMultipleChoice: false,
              difficulty: 200,
              isCompleted: false,
              isRequired: false,
              questionAnswers: [
                {
                  answerIndex: 0,
                  answerText: "testAnswerText",
                  isSelected: false,
                },
              ],
              hints: [
                {
                  hintID: 0,
                  showOnIsWrong: true,
                  hintAction: {
                    hintActionData: "testHintActionData",
                    hintActionType: AdaptivityElementActionTypes.CommentAction,
                  },
                },
              ],
            },
          ],
        }}
        setHeaderText={() => {}}
        onSelectQuestion={() => {}}
        onSelectHint={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("should render (every required combination)", () => {
    const { container } = render(
      <AdaptivityElementQuestionSelection
        selectedTask={{
          taskID: 0,
          taskTitle: "testTaskTitle",
          isRequired: true,
          isCompleted: false,
          requiredDifficulty: 0,
          questions: [
            {
              questionID: 0,
              questionText: "testQuestionText",
              isMultipleChoice: false,
              difficulty: 100,
              isCompleted: true,
              isRequired: true,
              questionAnswers: [
                {
                  answerIndex: 0,
                  answerText: "testAnswerText",
                  isSelected: false,
                },
              ],
              hints: [],
            },
            {
              questionID: 0,
              questionText: "testQuestionText",
              isMultipleChoice: false,
              difficulty: 200,
              isCompleted: false,
              isRequired: true,
              questionAnswers: [
                {
                  answerIndex: 0,
                  answerText: "testAnswerText",
                  isSelected: false,
                },
              ],
              hints: [],
            },
          ],
        }}
        setHeaderText={() => {}}
        onSelectQuestion={() => {}}
        onSelectHint={() => {}}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test("onSelectQuestion button calls onSelectQuestion callback when clicked", () => {
    const onSelectQuestionMock = jest.fn();
    const question = {
      questionID: 42,
      questionText: "testQuestionText",
      isMultipleChoice: false,
      difficulty: 0,
      isCompleted: false,
      isRequired: false,
      questionAnswers: [
        {
          answerIndex: 0,
          answerText: "testAnswerText",
          isSelected: false,
        },
      ],
      hints: [],
    };
    const { getByText } = render(
      <AdaptivityElementQuestionSelection
        selectedTask={{
          taskID: 0,
          taskTitle: "testTaskTitle",
          isCompleted: false,
          isRequired: false,
          requiredDifficulty: 0,
          questions: [question],
        }}
        setHeaderText={() => {}}
        onSelectQuestion={onSelectQuestionMock}
        onSelectHint={() => {}}
      />,
    );
    getByText("easy").click();

    expect(onSelectQuestionMock).toHaveBeenCalledTimes(1);
    expect(onSelectQuestionMock).toHaveBeenCalledWith(question);
  });

  // ANF-ID: [EWE0010]
  test("onSelectHint button calls onSelectHint callback when clicked", () => {
    const onSelectHintMock = jest.fn();
    const hint = {
      hintID: 0,
      showOnIsWrong: true,
      hintAction: {
        hintActionData: "testHintActionData",
        hintActionType: AdaptivityElementActionTypes.CommentAction,
      },
    };
    const question = {
      questionID: 42,
      questionText: "testQuestionText",
      isMultipleChoice: false,
      difficulty: 0,
      isCompleted: false,
      isRequired: false,
      questionAnswers: [
        {
          answerIndex: 0,
          answerText: "testAnswerText",
          isSelected: false,
        },
      ],
      hints: [hint],
    };
    const { getByText } = render(
      <AdaptivityElementQuestionSelection
        selectedTask={{
          taskID: 0,
          taskTitle: "testTaskTitle",
          isCompleted: false,
          isRequired: false,
          requiredDifficulty: 0,
          questions: [question],
        }}
        setHeaderText={() => {}}
        onSelectQuestion={() => {}}
        onSelectHint={onSelectHintMock}
      />,
    );
    getByText("hintButton").click();

    expect(onSelectHintMock).toHaveBeenCalledTimes(1);
    expect(onSelectHintMock).toHaveBeenCalledWith(hint, question);
  });
});
