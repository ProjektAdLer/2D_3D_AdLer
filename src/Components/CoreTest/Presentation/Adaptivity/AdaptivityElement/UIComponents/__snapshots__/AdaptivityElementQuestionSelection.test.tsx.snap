// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdaptivityElementQuestionSelection should render (every required combination) 1`] = `
<div>
  <div
    class="grid w-full gap-4 px-2 py-2"
  >
    <div
      class="flex items-center w-full gap-4 "
    >
      <div
        class=""
      >
        <button
          class="w-full col-span-2 md:col-span-1 px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
        >
          <div
            class="grid items-center w-full h-full grid-cols-5 "
          >
            <div
              class="flex items-center h-full"
            >
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt=""
                class="w-4 h-4 md:w-6 md:h-6"
                src="diffculties-adaptivity-medium-required-solved.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
            </div>
            <p
              class="flex justify-center col-span-3 px-4 text-sm text-start"
            >
              normal
               
            </p>
            <div
              class="flex items-center justify-end pr-2"
            >
              <img
                alt=""
                class="h-5 md:h-7"
                src="check-solution-icon-nobg.svg"
              />
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex flex-row h-full md:gap-4 "
      />
    </div>
    <div
      class="flex items-center w-full gap-4 "
    >
      <div
        class=""
      >
        <button
          class="w-full col-span-2 md:col-span-1 px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
        >
          <div
            class="grid items-center w-full h-full grid-cols-5 "
          >
            <div
              class="flex items-center h-full"
            >
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt=""
                class="w-4 h-4 md:w-6 md:h-6"
                src="diffculties-adaptivity-hard-required-tried.svg"
              />
            </div>
            <p
              class="flex justify-center col-span-3 px-4 text-sm text-start"
            >
              hard
               
            </p>
            <div
              class="flex items-center justify-end pr-2"
            >
              <img
                alt=""
                class="h-5 md:h-7"
                src="required-adaptivity.svg"
              />
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex flex-row h-full md:gap-4 "
      />
    </div>
  </div>
</div>
`;

exports[`AdaptivityElementQuestionSelection should render (with all difficulties) 1`] = `
<div>
  <div
    class="grid w-full gap-4 px-2 py-2"
  >
    <div
      class="flex items-center w-full gap-4 "
    >
      <div
        class=""
      >
        <button
          class="w-full col-span-2 md:col-span-1 px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
        >
          <div
            class="grid items-center w-full h-full grid-cols-5 "
          >
            <div
              class="flex items-center h-full"
            >
              <img
                alt=""
                class="w-4 h-4 md:w-6 md:h-6"
                src="diffculties-adaptivity-easy-unsolved-icon.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
            </div>
            <p
              class="flex justify-center col-span-3 px-4 text-sm text-start"
            >
              easy
               
            </p>
            <div
              class="flex items-center justify-end pr-2"
            >
              <div
                class="w-16"
              />
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex flex-row h-full md:gap-4 "
      />
    </div>
    <div
      class="flex items-center w-full gap-4 "
    >
      <div
        class=""
      >
        <button
          class="w-full col-span-2 md:col-span-1 px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
        >
          <div
            class="grid items-center w-full h-full grid-cols-5 "
          >
            <div
              class="flex items-center h-full"
            >
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt=""
                class="w-4 h-4 md:w-6 md:h-6"
                src="diffculties-adaptivity-medium-solved.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
            </div>
            <p
              class="flex justify-center col-span-3 px-4 text-sm text-start"
            >
              normal
               
            </p>
            <div
              class="flex items-center justify-end pr-2"
            >
              <div
                class="w-16"
              />
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex flex-row h-full md:gap-4 "
      >
        <div
          class=""
        >
          <button
            class="w-full h-full px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          >
            <p
              class="text-sm"
            >
              Hinweis
            </p>
          </button>
        </div>
      </div>
    </div>
    <div
      class="flex items-center w-full gap-4 "
    >
      <div
        class=""
      >
        <button
          class="w-full col-span-2 md:col-span-1 px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
        >
          <div
            class="grid items-center w-full h-full grid-cols-5 "
          >
            <div
              class="flex items-center h-full"
            >
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt="Platzhalter-Icon"
                class="w-4 h-4 md:w-6 md:h-6 opacity-60"
                src="diffculties-adaptivity-placeholder.svg"
              />
              <img
                alt=""
                class="w-4 h-4 md:w-6 md:h-6"
                src="diffculties-adaptivity-hard-tried.svg"
              />
            </div>
            <p
              class="flex justify-center col-span-3 px-4 text-sm text-start"
            >
              hard
               
            </p>
            <div
              class="flex items-center justify-end pr-2"
            >
              <div
                class="w-16"
              />
            </div>
          </div>
        </button>
      </div>
      <div
        class="flex flex-row h-full md:gap-4 "
      >
        <div
          class=""
        >
          <button
            class="w-full h-full px-2 py-1 justify-center flex items-center text-sm rounded-lg hover:cursor-pointer hover:border-buttonbgblue hover:text-buttonbgblue hover:bg-adlerdarkblue lg:text-xl transition ease-in-out duration-75 active:translate-x-[1px] active:translate-y-[1px] active:border-b-2 active:border-r-2 text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue overflow-hidden box-border cursor-pointer bg-buttonbgblue"
          >
            <p
              class="text-sm"
            >
              Hinweis
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
